km/km_exec.c

```diff
@@ -368,15 +368,32 @@ char** km_exec_build_argv(char* filename, char** argv, char** envp)
       nargv[0] = km_get_self_name();                               // km exec
       return nargv;
    }
-   pl_name = strdup(filename);                                    // not shebang
-   if ((nargv = calloc(1 + argc + 1, sizeof(char*))) == NULL) {   // km exec, cnt, and NULL
+   pl_name = strdup(filename);    // not shebang
+   int new_argc = 1 + argc + 1;   // km exec, cnt, and NULL
+   if (km_machine_init_params.vdev_name != NULL) {
+      new_argc++;
+   }
+   if ((nargv = calloc(new_argc, sizeof(char*))) == NULL) {   // km exec, cnt, and NULL
       km_infox(KM_TRACE_EXEC, "Couldn't allocate %ld bytes", (1 + argc + 1) * sizeof(char*));
       free(argv);
       return NULL;
    }
-   nargv[0] = km_get_self_name();   // km exec
-   nargv[1] = pl_name;
-   memcpy(nargv + 2, argv + 1, sizeof(char*) * (argc - 1));
+   int new_argc_index = 0;
+   nargv[new_argc_index++] = km_get_self_name();   // km exec
+   if (km_machine_init_params.vdev_name != NULL) {
+      int string_len = strlen("--virt-device=") + strlen(km_machine_init_params.vdev_name) + 1;
```

Comment:
Maybe make a preprocessor symbol for "--virt-device=" since we use it twice here.
It helps people know they are the same.